cogs/kattis.py

```diff
@@ -0,0 +1,92 @@
+import random
+import json
+
+import discord
+from discord.ext import commands
+from discord import app_commands
+from discord.app_commands import Choice
+
+class Kattis(commands.GroupCog, name="kattis"):
+    def __init__(self):
+        self.book_url = ""
+        self.easy_problems = []
+        self.medium_problems = []
+        self.hard_problems = []
+        self.all_problems = []
+        self.specific_problems = {}
+
+    async def cog_load(self):
+        await super().cog_load()
+        print("Kattis Cog Loaded")
+
+    async def load_data(self):
+        with open("data/kattis.json", "r", encoding='utf-8') as f:
+            self.book_url = json.load(f)["book"]
+
+        with open("data/problems.json", "r", encoding='utf-8') as f:
+            problems = json.load(f)
+            self.easy_problems = problems["easy"]
+            self.medium_problems = problems["medium"]
+            self.hard_problems = problems["hard"]
+            self.all_problems = self.easy_problems + self.medium_problems + self.hard_problems
+
+        with open("data/specific.json", "r", encoding='utf-8') as f:
+            self.specific_problems = json.load(f)
+
+    def get_kattis_url(self, problem):
+        return f"https://open.kattis.com/problems/{problem}"
+
+    @app_commands.command(name="book", description="Get the url to a random google sheets document idk.")
+    async def book(self, intr: discord.Interaction):
+        await intr.response.send_message(self.book_url)
+
+    @app_commands.command(name="problem", description="Get a random kattis problem based on the chosen difficulty.")
+    @app_commands.describe(difficulty="Prolem difficulty")
+    @app_commands.choices(difficulty=[
+        Choice(name="easy", value=0),
+        Choice(name="medium", value=1),
+        Choice(name="hard", value=2),
+        Choice(name="any", value=3)])
+    async def problem(self, intr: discord.Interaction, difficulty: Choice[int]):
+        problem = ""
+        match difficulty.value:
```

Comment:
nice match case, will need to check with @Sooraj-beep whether we have access to >=3.10 tho.

PS, if not, we could refactor to array indexing operation: random.choice([self.easy. self.medium][difficulty.value])